can/interfaces/robotell.py

```diff
@@ -0,0 +1,404 @@
+"""
+Interface for Chinese Robotell compatible interfaces (win32/linux).
+"""
+
+import time
+import logging
+
+from can import BusABC, Message
+
+logger = logging.getLogger(__name__)
+
+try:
+    import serial
+except ImportError:
+    logger.warning(
+        "You won't be able to use the Robotell can backend without "
+        "the serial module installed!"
+    )
+    serial = None
+
+
+class robotellBus(BusABC):
+    """
+    robotell interface
+    """
+
+    _PACKET_HEAD = 0xAA  # Frame starts with 2x FRAME_HEAD bytes
+    _PACKET_TAIL = 0x55  # Frame ends with 2x FRAME_END bytes
+    _PACKET_ESC = (
+        0xA5  # Escape char before any HEAD, TAIL or ESC chat (including in checksum)
+    )
+
+    _CAN_CONFIG_CHANNEL = 0xFF  # Configuration channel of CAN
+    _CAN_SERIALBPS_ID = 0x01FFFE90  # USB Serial port speed
+    _CAN_ART_ID = 0x01FFFEA0  # Automatic retransmission
+    _CAN_ABOM_ID = 0x01FFFEB0  # Automatic bus management
+    _CAN_RESET_ID = 0x01FFFEC0  # ID for initialization
+    _CAN_BAUD_ID = 0x01FFFED0  # CAN baud rate
+    _CAN_FILTER_BASE_ID = 0x01FFFEE0  # ID for first filter (filter0)
+    _CAN_FILTER_MAX_ID = 0x01FFFEE0 + 13  # ID for the last filter (filter13)
+    _CAN_INIT_FLASH_ID = 0x01FFFEFF  # Restore factory settings
+    _CAN_READ_SERIAL1 = 0x01FFFFF0  # Read first part of device serial number
+    _CAN_READ_SERIAL2 = 0x01FFFFF1  # Read first part of device serial number
+    _MAX_CAN_BAUD = 1000000  # Maximum supported CAN baud rate
+    _FILTER_ID_MASK = 0x0000000F  # Filter ID mask
+    _CAN_FILTER_EXTENDED = 0x40000000  # Enable mask
+    _CAN_FILTER_ENABLE = 0x80000000  # Enable filter
+
+    _CAN_STANDARD_FMT = 0  # Standard message ID
+    _CAN_EXTENDED_FMT = 1  # 29 Bit extended format ID
+    _CAN_DATA_FRAME = 0  # Send data frame
+    _CAN_REMOTE_FRAME = 1  # Request remote frame
+
+    def __init__(
+        self, channel, ttyBaudrate=115200, bitrate=None, rtscts=False, **kwargs
+    ):
+        """
+        :param str channel:
+            port of underlying serial or usb device (e.g. /dev/ttyUSB0, COM8, ...)
+            Must not be empty.
+        :param int ttyBaudrate:
+            baudrate of underlying serial or usb device
+        :param int bitrate:
+            CAN Bitrate in bit/s. Value is stored in the adapter and will be used as default if no bitrate is specified
+        :param float poll_interval:
+            Poll interval in seconds when reading messages
```

Comment:
You can remove this.

Reply:
Good point! I used the slcan driver as template and blindly copied this parameter description even though I don't use it. Removing.

Think we should also remove the parameter description from slcan.py - it does not seem to be used there either.